### PR TITLE
Fix `RouterPromptHandler` to not show a prompt when navigating to a path with params that is not a sub route

### PR DESCRIPTION
**Pull request by nsams**

### CODE REVIEW

File: packages/admin/admin/src/router/PromptHandler.tsx [hunk 1]
```diff
@@ -43,7 +43,11 @@ function InnerPromptHandler({
             const path = registeredMessages.current[id].path;
             const subRoutePath = registeredMessages.current[id].subRoutePath;
             // allow transition if location is below path where prompt was rendered
-            if (!((subRoutePath && location.pathname.startsWith(subRoutePath)) || location.pathname == path)) {
+            if (subRoutePath && location.pathname.startsWith(subRoutePath)) {
+                //subRoutePath matches with loaciton, allow transition
```
Comment (thomasdax98) [inline packages/admin/admin/src/router/PromptHandler.tsx:47]:
```suggestion
                //subRoutePath matches with location, allow transition
```

File: packages/admin/admin/src/router/PromptHandler.test.tsx [hunk 2]
```diff
@@ -51,6 +52,62 @@ test("Nested route in Prompt", async () => {
     });
 });
 
+test("Nested dynamic route in Prompt", async () => {
+    function Aasdf() {
```
Comment (johnnyomair) [inline packages/admin/admin/src/router/PromptHandler.test.tsx:56]: Come'on 😁

File: packages/admin/admin/src/router/PromptHandler.test.tsx [hunk 3]
```diff
@@ -51,6 +52,62 @@ test("Nested route in Prompt", async () => {
...
+            <Switch>
+                <Route path="/foo/:param">
+                    <Aasdf />
+                </Route>
+                <Redirect to="/foo/paramvalue" />
+            </Switch>
+        );
+    }
+
+    const rendered = render(
```
Comment (johnnyomair) [inline packages/admin/admin/src/router/PromptHandler.test.tsx:84]: Please use the `render` function introduced in #1441, which allows you to remove these default providers

**Review by johnnyomair — COMMENTED** [review]
I don't understand what exactly the fix is: All routes in the component `Aasdf` are in the sub route? Did the RouterPromptHandler show an incorrect prompt when used inside a route with params?

**Comment by nsams** [comment]
> I don't understand what exactly the fix is: All routes in the component Aasdf are in the sub route? Did the RouterPromptHandler show an incorrect prompt when used inside a route with params?

Problem lies here, when navigating back (`fireEvent.click(rendered.getByText("back")); //go back again, navigating to /foo/:param` in the test):

`} else if (matchPath(location.pathname, { path, exact: true })) {`

here `path` is `"/foo/:param"` and `location.pathname` is `/foo/paramvalue` which where compared using `==` previously

**Comment by johnnyomair** [comment]
So basically you're navigating to the SubRouteIndexRoute, and that caused a false positive prompt?

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**